subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/internal/task/lifecycle/TaskLifecycleListener.kt

```diff
@@ -0,0 +1,24 @@
+package com.avito.android.build_verdict.internal.task.lifecycle
+
+import org.gradle.api.Task
+import org.gradle.util.Path
+
+abstract class TaskLifecycleListener<in T> {
+
+    protected abstract val logs: MutableMap<Path, StringBuilder>
+
+    abstract fun beforeExecute(task: T)
+
+    protected abstract fun afterSucceedExecute(task: T)
+
+    protected abstract fun afterFailedExecute(task: T)
+
+    fun afterExecute(task: T, failure: Throwable?) {
+        if (failure == null) {
+            logs.remove(Path.path((task as Task).path))
+            afterSucceedExecute(task)
+        } else {
```

Comment:
Why don't we remove from logs here?

Reply:
because we will use them further

Reply:
Exactly. It means that the abstract class knows about the behavior of children. Ideally, it shouldn't rely on this.